Modify rules S131 S3562: CPP-3692 clarify difference and purpose (#984)
SonarSource/rspec: rules/S131/cfamily/rule.adoc

The requirement for a final ``++default++`` clause is defensive programming. The clause should either take appropriate action, or contain a suitable comment as to why no action is taken. When the ``++switch++`` covers all current values of an ``++enum++`` - and especially when it doesn't - a ``++default++`` case should still be used because there is no guarantee that the ``++enum++`` won't be extended.

Note that there is a more nuanced version of this rule: S3562.
Use S131 if you want to require a `++default++` case for every `++switch++`

> even if it already handles each value of an `++enum++`.

Reviewer: ```suggestion
even if it already handles all enumerators of an `++enum++`.
```

> Use S3562 rule if you want to ensure all reasonable cases are handled,
> and do not want to include a `++default++` case if you handle all values in an `++enum++`.

Reviewer: ```suggestion
Use S3562 rule if you want to require the inclusion of a `++default++`, even  case if all enumerators in an `++enum++` are handled.
```

Author: rephrased.
